pkg/logs/tekton_logging.go

```diff
@@ -580,10 +576,22 @@ func downloadLogFile(logsURL string, o *opts.CommonOptions) ([]byte, error) {
 	return logBytes, nil
 }
 
-func performProviderDownload(logsURL string) (*bufio.Scanner, error) {
-	provider, err := factory.NewBucketProviderFromTeamSettingsConfiguration(clients.NewFactory())
+func performProviderDownload(logsURL string, jxClient versioned.Interface, ns string) (*bufio.Scanner, error) {
+	provider, err := NewBucketProviderFromTeamSettingsConfiguration(jxClient, ns)
```

Comment:
Is the problem that jxClient is not passed or that the namespace is not passed. This is for sure better, just in the end`clients.NewFactory()` works as well, right? Unless you switch context I guess. Trying to understand how the bug manifests itself.

Reply:
the problem is that the factories (i.e. the `jxClient` and `ns` in this case are created with default values, not the ones that the command was called with. This happens if you programatically reuse this code and create a custom jxClient (e.g. with impersonation)